### PR TITLE
Add System.Net.Cache types to corefx

### LINKED ISSUE #11791: Port System.Net.Cache

### PR DESCRIPTION
**Pull request by stephentoub**
This adds the exposed types/members in System.Net.Cache to corefx.  It does not actually provide the caching logic that would be used like a type like HttpWebRequest to do caching.

cc: @davidsh, @cipop, @ericeil, @karelz 
Fixes #11791

### CODE REVIEW

File: src/System.Net.Requests/ref/System.Net.Requests.cs [hunk 1]
```diff
@@ -124,3 +124,59 @@ public abstract partial class WebResponse : System.IDisposable
...
+namespace System.Net.Cache
+{
+    public enum HttpCacheAgeControl
+    {
+        None = 0x0,
+        MinFresh = 0x1,
+        MaxAge = 0x2,
+        MaxStale = 0x4,
+        MaxAgeAndMinFresh = 0x3,
+        MaxAgeAndMaxStale = 0x6,
```
Comment (davidsh) [inline src/System.Net.Requests/ref/System.Net.Requests.cs]: nit: comma after last item not needed

File: src/System.Net.Requests/ref/System.Net.Requests.cs [hunk 2]
```diff
@@ -124,3 +124,59 @@ public abstract partial class WebResponse : System.IDisposable
...
+    {
+        Default = 0,
+        BypassCache = 1,
+        CacheOnly = 2,
+        CacheIfAvailable = 3,
+        Revalidate = 4,
+        Reload = 5,
+        NoCacheNoStore = 6,
+        CacheOrNextCacheOnly = 7,
+        Refresh = 8,
```
Comment (davidsh) [inline src/System.Net.Requests/ref/System.Net.Requests.cs]: nit: comma after last item not needed

File: src/System.Net.Requests/src/System/Net/Cache/HttpCacheAgeControl.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+namespace System.Net.Cache
+{
+    public enum HttpCacheAgeControl
+    {
+        None = 0x0,
+        MinFresh = 0x1,
+        MaxAge = 0x2,
+        MaxStale = 0x4,
+        MaxAgeAndMinFresh = 0x3,
+        MaxAgeAndMaxStale = 0x6,
```
Comment (davidsh) [inline src/System.Net.Requests/src/System/Net/Cache/HttpCacheAgeControl.cs]: nit: comma after last item not needed

File: src/System.Net.Requests/src/System/Net/Cache/HttpRequestCacheLevel.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    {
+        Default = 0,
+        BypassCache = 1,
+        CacheOnly = 2,
+        CacheIfAvailable = 3,
+        Revalidate = 4,
+        Reload = 5,
+        NoCacheNoStore = 6,
+        CacheOrNextCacheOnly = 7,
+        Refresh = 8,
```
Comment (davidsh) [inline src/System.Net.Requests/src/System/Net/Cache/HttpRequestCacheLevel.cs]: nit: comma after last item not needed

File: src/System.Net.Requests/ref/System.Net.Requests.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Net.Requests/ref/System.Net.Requests.cs]: Removed

File: src/System.Net.Requests/ref/System.Net.Requests.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Net.Requests/ref/System.Net.Requests.cs]: Removed

File: src/System.Net.Requests/src/System/Net/Cache/HttpCacheAgeControl.cs (hunk 3 above)
Comment (stephentoub) [inline src/System.Net.Requests/src/System/Net/Cache/HttpCacheAgeControl.cs]: Removed

File: src/System.Net.Requests/src/System/Net/Cache/HttpRequestCacheLevel.cs (hunk 4 above)
Comment (stephentoub) [inline src/System.Net.Requests/src/System/Net/Cache/HttpRequestCacheLevel.cs]: Removed

**Comment by stephentoub** [comment]
@dotnet-bot Test Innerloop Ubuntu14.04 Debug Build and Test please (known hang due to OS)
@dotnet-bot Test Innerloop Windows_NT Release Build and Test please (https://github.com/dotnet/corefx/issues/11987)

File: src/System.Net.Requests/ref/System.Net.Requests.cs [hunk 5]
```diff
@@ -124,3 +124,59 @@ public abstract partial class WebResponse : System.IDisposable
         public abstract System.IO.Stream GetResponseStream();
     }
 }
+namespace System.Net.Cache
```
Comment (weshaggard) [inline src/System.Net.Requests/ref/System.Net.Requests.cs]: This change broken consumption of the new packages with this change. You cannot add more APIs into the older netstandard1.3 version. See http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/master/job/windows_nt_debug_prtest/543/consoleFull

This needs to be bumped to netstandard1.7. 

cc @stephentoub @ericstj

**Comment by stephentoub** [comment]
I'm confused. I thought that if we were only adding existing APIs and not taking new dependencies that we only needed to bump the dir.props value. And that if we got it wrong package validation would flag it. We should just go through and move every assembly to 1.7 or whatever. This piecemeal approach is causing way more headache and work than it's worth.

File: src/System.Net.Requests/ref/System.Net.Requests.cs (hunk 5 above)
Comment (ericstj) [inline src/System.Net.Requests/ref/System.Net.Requests.cs]: No it does not.  Those failures you are pointing out would be hit if that XML project was targeting netstandard 1.7 just the same.  The project is defining types that are now being exposed.

**Comment by ericstj** [comment]
If you move every assembly to netstandard1.7 you're just creating more pain since you'll be cross compiling for everything.

Rules should be as follows:
1. If package is split adding API requires bumping netstandard version.
  - reasoning here is that split packages do not pull in new implementations for portable apps, instead the runtime-specific implementation packages always come from the shared framework.
2. If package is not split, adding API can be done without bumping netstandard version, so long as all frameworks that support that version can support the new API.

I believe 2 is the case here, so what Steven has done is reasonable.

Sure it has fallout in the Xml project.  Perhaps the problem here is that our build process didn't let Steve discover that as part of hit commit.

**Pull request merged**